packages/esm-patient-search-app/src/types/index.ts

```diff
@@ -76,3 +77,35 @@ export interface PatientSearchResponse {
     }>[]
   >;
 }
+
+export interface AdvancedPatientSearchState {
+  gender: 'any' | 'male' | 'female';
+  dateOfBirth: number;
+  monthOfBirth: number;
+  yearOfBirth: number;
+  phoneNumber: number;
+  postcode: string;
+  age: number;
+}
+
+export enum AdvancedPatientSearchActionTypes {
+  SET_GENDER = 'SET_GENDER',
+  SET_DATE_OF_BIRTH = 'SET_DATE_OF_BIRTH',
+  SET_MONTH_OF_BIRTH = 'SET_MONTH_OF_BIRTH',
+  SET_YEAR_OF_BIRTH = 'SET_YEAR_OF_BIRTH',
+  SET_PHONE_NUMBER = 'SET_PHONE_NUMBER',
+  SET_POSTCODE = 'SET_POSTCODE',
+  SET_AGE = 'SET_AGE',
+  RESET_FIELDS = 'RESET_FIELDS',
+}
```

Comment:
Why not just use typescript? `searchActionType = "SET_GENDER" | "SET_DATE_OF_BIRTH" | ... `

Reply:
Just thought of going through this, will update it.
Thanks!